Makefile

```diff
@@ -57,10 +61,36 @@ test: generate $(ALL_GO_MODS:%=test/%)
 test/%/go.mod:
 	@cd $* && $(GOCMD) test $(ARGS) ./...
 
-.PHONY: generate
-generate: export CFLAGS := $(BPF_INCLUDE)
-generate: go-mod-tidy
-generate:
+PROBE_ROOT = internal/pkg/instrumentation/bpf/
+PROBE_DIRS = $(shell find $(PROBE_ROOT) -type d -exec sh -c '(ls -p "{}"|grep />/dev/null)||dirname "{}"' \;)
+PROBE_GEN := $(addsuffix /bpf_x86_bpfel.go,$(PROBE_DIRS))
+PROBE_GEN += $(addsuffix /bpf_arm64_bpfel.go,$(PROBE_DIRS))
+PROBE_GEN := $(wildcard $(PROBE_GEN))
+PROBE_GEN += internal/pkg/instrumentation/bpf/net/http/client/bpf_no_tp_arm64_bpfel.go
+PROBE_GEN += internal/pkg/instrumentation/bpf/net/http/client/bpf_no_tp_x86_bpfel.go
+
+# Include all depinfo files to ensure we only re-generate when needed.
+-include $(shell find $(PROBE_ROOT) -type f -name 'bpf_*_bpfel.go.d')
+
+$(PROBE_GEN): %_bpfel.go: %_bpfel.o
+
+%/bpf_x86_bpfel.o: generate/%
+	$(info building $@)
+
+%/bpf_arm64_bpfel.o: generate/%
+	$(info building $@)
+
+%/bpf_no_tp_x86_bpfel.o: generate/%
+	$(info building $@)
+
+%/bpf_no_tp_arm64_bpfel.o: generate/%
+	$(info building $@)
```

Comment:
If in the future we'll add another compilation variant we will need to add another target here. Is there a way to capture all the relevant files in one target (using regex)?

Reply:
I mean, sure, we can match all object files. Just, that will possibly also cause conflict with targets we don't want to match in the future. It seems like an issue that will need to be addressed in the future either way and being explicit seemed the more conservative approach.

I can change to the permissive case if that is desired though.